File: edukate-messaging/src/main/java/io/github/sanyavertolet/edukate/messaging/RabbitTopology.java
package io.github.sanyavertolet.edukate.messaging;

import lombok.experimental.UtilityClass;

@UtilityClass
public final class RabbitTopology {
    public static final String EXCHANGE = "edukate.exchange";

    public static final String RK_SCHEDULE = "edukate.check.schedule.v1";
    public static final String Q_SCHEDULE_CHECKER = "checker.check.schedule.v1.q";

    public static final String RK_RESULT   = "edukate.check.result.v1";
    public static final String Q_RESULT_BACKEND   = "backend.check.result.v1.q";

    public static final String RK_NOTIFY = "edukate.notify.v1";
    public static final String Q_NOTIFY  = "notifier.notify.v1.q";
}

File: edukate-messaging/src/main/kotlin/io/github/sanyavertolet/edukate/messaging/RabbitTopology.kt
package io.github.sanyavertolet.edukate.messaging

object RabbitTopology {
    const val EXCHANGE: String = "edukate.exchange"

    object Rk {
        const val SCHEDULE: String = "edukate.check.schedule.v1"
        const val RESULT: String = "edukate.check.result.v1"
        const val NOTIFY: String = "edukate.notify.v1"
    }

    object Q {
        const val SCHEDULE_CHECKER: String = "checker.check.schedule.v1.q"
        const val RESULT_BACKEND: String = "backend.check.result.v1.q"
        const val NOTIFY: String = "notifier.notify.v1.q"
    }
}
